src/main/java/seedu/address/model/tut/Tutorial.java

```diff
@@ -11,182 +11,402 @@
+        @Override
+        public List<Student> getStudents() {
+            return this.students;
+        }
 
-    }
+        @Override
+        public boolean setAttendance(Date date, StudentId target) {
+            requireNonNull(date);
+            requireNonNull(target);
+            TutDate tutDate = tutDates.computeIfAbsent(date, TutDate::new);
+            return students.stream()
+                    .filter(s -> s.getStudentId().equals(target))
+                    .findFirst()
+                    .map(student -> {
+                        student.setAttendance(tutDate);
+                        tutDate.add(target);
+                        return true;
+                    }).orElse(false);
+        }
 
-    @Override
-    public String toString() {
-        return tutName + ": Tutorial " + tutorialClass;
+        @Override
+        public void assignStudent(Student student) {
+            requireNonNull(student);
+
+            students.add(student);
+        }
+
+        @Override
+        public TutName getTutName() {
+            return this.tutName;
+        }
+
+        @Override
+        public Student get(Name name) {
+            return students.stream()
+                    .filter(student -> student.getName().equals(name))
+                    .findFirst()
+                    .orElse(null); // Returns null if no student is found
+        }
+
+        @Override
+        public boolean tutorialDateInList(TutDate date) {
+            return tutDates.containsKey(date.getDate());
+        }
+
+        @Override
+        public void addTutorialDate(TutDate tutorialDate) {
+            requireNonNull(tutorialDate);
+            tutDates.putIfAbsent(tutorialDate.getDate(), tutorialDate);
+        }
+
+        @Override
+        public TutDate getTutorialDate(Date date) {
+            return tutDates.get(date);
+        }
+
+        @Override
+        public boolean isValidTutorialDate(TutDate tutorialDate) {
+            return tutorialDate.isValid();
+        }
+
+        @Override
+        public List<TutDate> getTutDates() {
+            return new ArrayList<>(tutDates.values());
+        }
+
+        @Override
+        public TutorialClass getTutorialClass() {
+            return tutorialClass;
+        }
+
+        @Override
+        public void markAttendance(Student student, TutDate tutorialDate) throws TutDateNotFoundException {
+            if (!isValidTutorialDate(tutorialDate)) {
+                throw new TutDateNotFoundException();
+            }
+            if (!studentInList(student)) {
+                add(student);
+            }
+            addTutorialDate(tutorialDate);
+            tutorialDate.add(student.getStudentId());
+        }
+
+        @Override
+        public boolean studentInList(Student student) {
+            return students.contains(student);
+        }
+
+        @Override
+        public void deleteStudent(Student student) {
+            students.remove(student);
```

Comment:
Should this method throw studentNotFound exception if the student is not in the list? Just to remind the user in case there is some typo.